python/nav/web/alertprofiles/forms.py

```diff
@@ -625,3 +641,44 @@ def __init__(self, *args, **kwargs):
 
                 # At last we acctually add the multiple choice field.
                 self.fields['value'] = forms.MultipleChoiceField(choices=choices)
+            else:
+                self.fields['value'] = forms.CharField(required=True)
+
+    def clean(self) -> Dict[str, Any]:
+        validated_data = super().clean()
+
+        match_field = validated_data["match_field"]
+        operator_type = int(validated_data["operator"])
+        value = validated_data["value"]
+
+        if match_field.data_type == MatchField.IP:
+            if operator_type == Operator.IN:
+                ip_list = value.split()
+            else:
+                ip_list = [value]
+            validated_ip_addresses = []
+            for ip in ip_list:
+                try:
+                    IP(ip)
```

Comment:
Actually, I would advise against using `IPy.IP` for validation of strings, precisely because this library is extremely lenient in what is considered acceptable as input.

Case in point:

```pycon
>>> from IPy import IP
>>> IP(42)
IP('0.0.0.42')
>>> IP('42')
IP('42.0.0.0')
>>> IP(23948234)
IP('1.109.107.202')
>> IP('100.200')
IP('100.200.0.0')
>>>
```

Please consider using `nav.utils.is_valid_ip()` instead (possibly with the `strict` flag set).